django/db/models/expressions.py

```diff
@@ -889,6 +893,30 @@ def as_sql(self, compiler, connection, template=None, **extra_context):
         template = template or self.template
         return (template % placeholders).rstrip(), params
 
+    def as_sqlite(self, compiler, connection):
+        template = None
+        if self.nullslast:
+            template = '%(expression)s IS NULL, %(expression)s %(ordering)s'
+        sql, params = self.as_sql(compiler, connection, template=template)
+        return sql, params
+
+    def as_postgresql(self, compiler, connection):
```

Comment:
It'd be better to rename this `as_sql` and to remove the `as_oracle` method since it is the "default" implementation that all backends follow unless they opt out. As-is, any third party backends *must* implement their own version even if they support NULLS LAST syntax, such as mssql.